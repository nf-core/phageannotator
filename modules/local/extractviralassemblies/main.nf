process EXTRACTVIRALASSEMBLIES {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(virus_fasta_gz)

    output:
    tuple val(meta), path("${prefix}_viral_assemblies.tsv") , emit: viral_assemblies
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_viral_assemblies.py \\
        --virus_fasta ${virus_fasta_gz} \\
        --output ${prefix}_viral_assemblies.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_viral_assemblies.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """
}
