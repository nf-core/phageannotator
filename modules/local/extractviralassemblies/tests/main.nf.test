nextflow_process {

    name "Test process: EXTRACTVIRALASSEMBLIES"
    script "../main.nf"
    process "EXTRACTVIRALASSEMBLIES"

    tag "modules"
    tag "modules_local"
    tag "extractviralassemblies"


    test("fasta.gz") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                        file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkifExists: true),
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                        file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkifExists: true),
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
