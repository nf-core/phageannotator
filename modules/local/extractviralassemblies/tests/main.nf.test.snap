{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "versions": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "viral_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "timestamp": "2024-04-03T20:03:31.598684897"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d98e022dc9c925d607bbbce2a5e62055"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "versions": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "viral_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d98e022dc9c925d607bbbce2a5e62055"
                    ]
                ]
            }
        ],
        "timestamp": "2024-04-03T20:03:15.882889502"
    }
}